ui/desktop/src/components/ChatView.tsx

```diff
@@ -94,6 +100,101 @@ export default function ChatView({
+        'Based on our conversation so far, could you create:\n' +
+        "1. A concise set of instructions (1-2 paragraphs) that describe what you've been helping with. Pay special attention if any output styles or formats are requested (and make it clear), and note any non standard tools used or required.\n" +
+        '2. A list of 3-5 example activities (as a few words each at most) that would be relevant to this topic\n\n' +
+        "Format your response with clear headings for 'Instructions:' and 'Activities:' sections.";
+
+      // Set waiting state to true before adding the prompt
+      setWaitingForAgentResponse(true);
+
+      // Add the prompt as a user message
+      append(createUserMessage(instructionsPrompt));
+
+      window.electron.logInfo('Injected instructions prompt into chat');
+    };
+
+    window.addEventListener('make-agent-from-chat', handleMakeAgent);
+
+    return () => {
+      window.removeEventListener('make-agent-from-chat', handleMakeAgent);
+    };
+  }, [append, chat.messages, setWaitingForAgentResponse]);
+
+  // Listen for new messages and process agent response
+  useEffect(() => {
+    // Only process if we're waiting for an agent response
+    if (!waitingForAgentResponse || messages.length === 0) {
+      return;
+    }
+
+    // Get the last message
+    const lastMessage = messages[messages.length - 1];
+
+    // Check if it's an assistant message (response to our prompt)
+    if (lastMessage.role === 'assistant') {
+      // Extract the content
+      const content = getTextContent(lastMessage);
+
+      // Process the agent's response
+      if (content) {
+        window.electron.logInfo('Received agent response:');
+        window.electron.logInfo(content);
+
+        // Parse the response to extract instructions and activities
+        const instructionsMatch = content.match(/Instructions:(.*?)(?=Activities:|$)/s);
+        const activitiesMatch = content.match(/Activities:(.*?)$/s);
+
+        const instructions = instructionsMatch ? instructionsMatch[1].trim() : '';
+        const activitiesText = activitiesMatch ? activitiesMatch[1].trim() : '';
+
+        // Parse activities into an array
+        const activities = activitiesText
+          .split(/\n+/)
+          .map((line) => line.replace(/^[•\-*\d]+\.?\s*/, '').trim())
+          .filter((activity) => activity.length > 0);
+
+        // Create a bot config object
+        const generatedConfig = {
+          id: `bot-${Date.now()}`,
+          name: 'Custom Bot',
+          description: 'Bot created from chat',
+          instructions: instructions,
+          activities: activities,
+        };
+
+        window.electron.logInfo('Extracted bot config:');
+        window.electron.logInfo(JSON.stringify(generatedConfig, null, 2));
+
+        // Store the generated bot config
+        setGeneratedBotConfig(generatedConfig);
+
+        // Show the modal with the generated bot config
+        setShowDeepLinkModal(true);
```

Comment:
nit: name should probably be about shareable bot?

Reply:
good point